src/frontend/src/lib/utils/token-group.utils.ts

```diff
@@ -164,3 +165,52 @@ export const groupMainToken = ({ token, tokenGroup }: GroupTokenParams): TokenUi
  */
 export const groupSecondaryToken = ({ token, tokenGroup }: GroupTokenParams): TokenUiGroup =>
 	nonNullish(tokenGroup) ? updateTokenGroup({ token, tokenGroup }) : mapNewTokenGroup(token);
+
+/**
+ * Function to create a list of TokenUiGroup by grouping a provided list of tokens.
+ *
+ * The function loops through the tokens list and groups them according to a prop key that links a token to its "main token".
+ * For example, it could be `twinToken` as per ck token standard. But the logic can be extended to other props, if necessary.
+ *
+ * The tokens with no "main token" will still be included in a group, but it will be a single-element group, where the "main token" is the token itself.
+ * That, in general, makes sense for tokens that are not "secondary tokens" of a "main token".
+ *
+ * The returned list respects the sorting order of the initial tokens list, meaning that the group is created at each position of the first encountered token of the group.
+ * So, independently of being a "main token" or a "secondary token", the group will replace the first token of the group in the list.
+ * That is useful if a "secondary token" is before the "main token" in the list; for example, if the list is sorted by balance.
+ *
+ * NOTE: The function does not sort the groups by any criteria. It only groups the tokens. So, even if a group ends up having a total balance that would put it in a higher position in the list, it will not be moved.
+ *
+ * @param {TokenUi[]} tokens - The list of TokenUi objects to group. Each token may or may not have a prop key to identify a "main token".
+ * @returns {TokenUiGroup[]} A list where tokens are grouped into a TokenUiGroup, even if they are by themselves.
+ */
+export const groupTokens = (tokens: TokenUi[]): TokenUiGroup[] => {
+	const tokenGroupsMap = tokens.reduce<{
+		[id: TokenId]: TokenUiGroup | undefined;
+	}>(
+		(acc, token) => ({
+			...acc,
+			...('twinToken' in token &&
+			isToken(token.twinToken) &&
+			token.decimals === token.twinToken.decimals
```

Comment:
Not necesseraly for this PR but the comparison of decimals feels like something important so maybe worth a standalone function for it?